extensions/amp-story/1.0/amp-story.js

```diff
@@ -1281,17 +1310,53 @@ export class AmpStory extends AMP.BaseElement {
       return Promise.resolve();
     }
 
+    const subscriptionsState = this.storeService_.get(
+      StateProperty.SUBSCRIPTIONS_STATE
+    );
     if (
-      isExperimentOn(this.win, 'amp-story-paywall-exp') &&
-      targetPage.isPaywallProtected()
+      isExperimentOn(this.win, 'amp-story-subscriptions') &&
+      this.isPaywallStory_()
     ) {
-      if (this.storeService_.get(StateProperty.SUBSCRIPTIONS_DIALOG_STATE)) {
-        // Subscription dialog is already triggered.
-        return Promise.resolve();
+      if (
+        pageIndex >= PAYWALL_PAGE_INDEX &&
+        subscriptionsState !== SubscriptionsState.GRANTED
+      ) {
+        // Hit a blocked page.
+        if (subscriptionsState === SubscriptionsState.UNKNOWN) {
```

Comment:
Isn't `UNKNOWN` the same as `this.pendingSubscriptionsState_`?

Reply:
In a situation where already waiting for entitlement, multiple clicks would result in multiple switchTo operations. This is to ensure the callback below would only executed once.

Reply:
We should block all operations while pending, so I dont think they can re-enter

Reply:
Actually, if user taps right multiple times while the sub state is pending, the switchTo would be called multiple times without this pending state check. That said, I find that switchTo itself already has some guard against multiple switchTo calls to the same page at line 1304. So should be fine without the pending state check. Removed.